docs/specs/thrid-party-client-draft.md

```diff
@@ -0,0 +1,38 @@
+## Third-party Clients
+
+The purpose of this document is to shape the third-party client feature. After the details become concrete, this document will be removed and merged to [OIDC](./oidc.md).
+
+## Authorization Flow
+
+- Remove `is_first_party` flag in oauth client, all existing OAuth clients should be first-party apps. Instead, adding `third_party_app` as new `x_application_type`.
```

Comment:
Worth mentioning how we migrate this. This is what I have in mind

1. Treat all client as first party, ignore is_first_party. Make is_first_party not required in the JSON schema. Mark this as VersionA
2. Deploy VersionA
3. Add and run a config migration to remove is_first_party.
4. Remove is_first_party from the JSON schema. Mark this as VersionB
5. Deploy VersionB